From 30913acff627ca0d71a503e55a91a0258b3b224f Mon Sep 17 00:00:00 2001
From: Daniel Leung <daniel.leung@intel.com>
Date: Fri, 9 Nov 2012 13:05:16 -0800
Subject: wifi: Add ability to use custom wifi code

Adds a switch so that a different source file for wifi can be
used instead of the default one from AOSP. This gives us the
flexibility of having different wifi code for various build
targets.

One may argue that the code can be built as a static library
and be included here. However, since nothing in here refers
to functions in the wifi code, the linker simply does not
include the library. So the source has to be built at the
same time as other source files here.

Also by moving the code away from this project, this makes
it easier for us to move forward to upstream commits.

For: AXIA-1242
Change-Id: Icf3851590742788c23eca12ca3344515ea6f1fcf
Signed-off-by: Daniel Leung <daniel.leung@intel.com>
---
 wifi/Android.mk |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/wifi/Android.mk b/wifi/Android.mk
index 7f06c51..85d4e7a 100644
--- a/wifi/Android.mk
+++ b/wifi/Android.mk
@@ -3,6 +3,10 @@
 LOCAL_CFLAGS += -DCONFIG_CTRL_IFACE_CLIENT_DIR=\"/data/misc/wifi/sockets\"
 LOCAL_CFLAGS += -DCONFIG_CTRL_IFACE_CLIENT_PREFIX=\"wpa_ctrl_\"
 
+ifneq ($(BOARD_CUSTOM_WIFI_HAL_SRC),)
+LOCAL_SRC_FILES += ../../$(BOARD_CUSTOM_WIFI_HAL_SRC)
+else
+
 ifdef WIFI_DRIVER_MODULE_PATH
 LOCAL_CFLAGS += -DWIFI_DRIVER_MODULE_PATH=\"$(WIFI_DRIVER_MODULE_PATH)\"
 endif
@@ -33,4 +37,6 @@ endif
 
 LOCAL_SRC_FILES += wifi/wifi.c
 
+endif
+
 LOCAL_SHARED_LIBRARIES += libnetutils
-- 
1.7.4.1

